pkg/platform/packet/packet.go

```diff
@@ -237,6 +255,79 @@ func (c *config) terraformSmartApply(ex *terraform.Executor, dnsProvider dns.DNS
 	return ex.Apply()
 }
 
+// terraformAddDependencies adds explicit dependencies to cluster nodes so nodes
+// with a specific hw reservation UUID are created before nodes that don't have
+// a specific hw reservation UUID.
+// The function modifies c.NodesDependsOn and c.workerPools[].NodesDependsOn,
+// assigning to them a string, formatted as a terraform list, of all worker
+// terraform targets needed to be created first. For example:
+//
+//	// Suppose worker pool "example" is the only to use a specific hw
+//	// reservation IDs. IOW, the controllers (c.NodesDependsOn) depends on
+//	// worker pool "example" to be created first.
+//	// Then, after calling this function, the attribute will be:
+// 	c.NodesDependsOn = "[ module.worker-example.worker_nodes_ids ]"
+//
+//
+// The explicit terraform dependency is needed to guarantees that nodes using
+// hardware reservation "next-available" won't use reservation IDS that another
+// worker pool may specify with a specific UUID, and thus fail to create the
+// node. This race condition is best explained here, if you want more info:
+// https://github.com/terraform-providers/terraform-provider-packet/issues/176
+// https://github.com/terraform-providers/terraform-provider-packet/pull/208
+func (c *config) terraformAddDependencies() {
```

Comment:
It would be nice to have some tests to validate the output of it.

Reply:
Makes sense! Created a list of pending things in the PR description :)

Reply:
Marking as solved, to not double-track things :)